services-api/src/main/java/io/scalecube/services/api/ServiceMessage.java

```diff
@@ -31,13 +31,15 @@
   public static final String HEADER_DATA_FORMAT = "_data_format";
 
   private Map<String, String> headers = new HashMap<>(1);
+  private Object credentials;
```

Comment:
Let's create a base class for it. I see it as misuse of `Object`.

Reply:
Credentials are part of headers now.